pkg/redirectpolicy/manager.go

```diff
@@ -46,8 +47,12 @@ type svcCache interface {
 	GetServiceFrontendIP(svcID k8s.ServiceID, svcType lb.SVCType) net.IP
 }
 
-// podID is pod name and namespace
-type podID = k8s.ServiceID
+// podID is a unique identifier for pods.
+type podID = struct {
+	k8s.ServiceID
+
+	uid types.UID
```

Comment:
Careful here. If a pod is deleted and added with the same name, it is possible for the shared informer to coalesce this in to a single update.

As a general rule, unless we really need to associate with the specific "version" of a pod, we should ignore UIDs.

Reply:
https://github.com/cilium/cilium/pull/31605#issuecomment-2097483504